crates/symbolicator/src/services/symbolication/process_minidump.rs

```diff
@@ -0,0 +1,1227 @@
+    ///
+    /// This function will load the CFI files and ask breakpad to stackwalk the minidump.
+    /// Once it has stacktraces it creates the list of used modules and returns the
+    /// un-symbolicated stacktraces in a structure suitable for requesting symbolication.
+    ///
+    /// The module list returned is usable for symbolication itself and is also directly
+    /// used in the final symbolication response of the public API.  It will contain all
+    /// modules which either have been referenced by any of the frames in the stacktraces or
+    /// have a full debug id.  This is intended to skip over modules like `mmap`ed fonts or
+    /// similar which are mapped in the address space but do not actually contain executable
+    /// modules.
+    #[tracing::instrument(skip_all)]
+    async fn stackwalk_minidump(
+        &self,
+        path: &Path,
+        cfi_caches: &CfiCacheModules,
+        return_modules: bool,
+    ) -> anyhow::Result<StackWalkMinidumpResult> {
+        let pool = self.spawnpool.clone();
+        let cfi_caches = cfi_caches.for_processing();
+        let minidump_path = path.to_path_buf();
+
+        let lazy = async move {
+            let spawn_time = std::time::SystemTime::now();
+            let spawn_result = pool.spawn(
+                (
+                    procspawn::serde::Json(cfi_caches.clone()),
+                    minidump_path.clone(),
+                    spawn_time,
+                    return_modules,
+                ),
+                |(cfi_caches, minidump_path, spawn_time, return_modules)| -> Result<_, ProcError> {
+                    let procspawn::serde::Json(cfi_caches) = cfi_caches;
+                    let rt = tokio::runtime::Builder::new_current_thread()
+                        .enable_all()
+                        .build()
+                        .unwrap();
+                    rt.block_on(async move {
+                        stackwalk_with_rust_minidump(
+                            cfi_caches,
+                            minidump_path,
+                            spawn_time,
+                            return_modules,
+                        )
+                        .await
+                        .map(procspawn::serde::Json)
+                    })
+                },
+            );
+
+            let result = Self::join_procspawn(
+                spawn_result,
+                Duration::from_secs(60),
+                "minidump.stackwalk.spawn.error",
+            )?;
+
+            metric!(timer("minidump.stackwalk.duration") = result.duration);
+
+            Ok::<_, anyhow::Error>(result)
+        };
+
+        self.cpu_pool
+            .spawn(lazy.bind_hub(sentry::Hub::current()))
+            .await
+            .context("Minidump stackwalk future cancelled")?
+    }
+
+    /// Iteratively stackwalks/processes the given `minidump_file` using breakpad.
```

Comment:
nit: i assume the breakpad bit of this is irrelevant? we can replace it with rust-minidump as a quick fix, or maybe provide a brief explanation on why there's a need for an iterative version of stackwalking.